src/main/java/com/maxmind/db/Networks.java

```diff
@@ -0,0 +1,184 @@
+     */
+    Networks(
+            Reader reader,
+            boolean skipAliasedNetworks,
+            NetworkNode[] nodes,
+            Class<T> typeParameterClass)
+        throws ClosedDatabaseException {
+        this.reader = reader;
+        this.skipAliasedNetworks = skipAliasedNetworks;
+        this.buffer = reader.getBufferHolder().get();
+        this.nodes = new Stack<NetworkNode>();
+        this.typeParameterClass = typeParameterClass;
+        for (NetworkNode node : nodes) {
+            this.nodes.push(node);
+        }
+    }
+
+    /**
+     * Constructs a Networks instance with skipAliasedNetworks set to false by default.
+     * @param reader The reader object.
+     * @param typeParameterClass The type of data returned by the iterator.
+     */
+    Networks(Reader reader, Class<T> typeParameterClass) throws ClosedDatabaseException {
+        this(reader, false, typeParameterClass);
+    }
+
+    /**
+     * Returns the next DataRecord. You need to set the class using
+     * prepareForClass before calling next. 
+     * For example,
+     *  networks.prepareForClass(Map.Class);
+     *  Map test = networks.next();
+     */
+    @Override
+    public DatabaseRecord<T> next() {
+        try {
+            T data = this.reader.resolveDataPointer(
+                this.buffer, this.lastNode.pointer, this.typeParameterClass);
+
+            byte[] ip = this.lastNode.ip;
+            int prefixLength = this.lastNode.prefix;
+
+            // We do this because uses of SkipAliasedNetworks expect the IPv4 networks
+            // to be returned as IPv4 networks. If we are not skipping aliased
+            // networks, then the user will get IPv4 networks from the ::FFFF:0:0/96
+            // network.
+            if (this.skipAliasedNetworks && isInIpv4Subtree(ip)) {
+                ip = Arrays.copyOfRange(ip, 12, ip.length);
+                prefixLength -= 96;
+            }
+
+            // If the ip is in ipv6 form, drop the prefix manually
+            // as InetAddress converts it to ipv4.
+            InetAddress ipAddr = InetAddress.getByAddress(ip);
+            if (ipAddr instanceof Inet4Address && ip.length > 4 && prefixLength > 96) {
+                prefixLength -= 96;
+            }
+
+            return new DatabaseRecord<T>(data, InetAddress.getByAddress(ip), prefixLength);
+        } catch (IOException e) {
+            throw new NetworksIterationException(e);
+        }
+    }
+
+    private boolean isInIpv4Subtree(byte[] ip) {
+        if (ip.length != 16) {
+            return false;
+        }
+        for (int i = 0; i < 12; i++) {
+            if (ip[i] != 0) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    /*
+    * Next prepares the next network for reading with the Network method. It
+    * returns true if there is another network to be processed and false if there
+    * are no more networks or if there is an error.
```

Comment:
"or if there is an error" can probably be removed now.

I think we should document that it throws `NetworksIterationException`.